app/conversations/document_based.py

```diff
@@ -18,14 +37,16 @@ def __init__(self):
         """
         self.text_splitter = RecursiveCharacterTextSplitter(
             chunk_size=500, chunk_overlap=20, length_function=len)
+        self.llama = get_llama_model()
+        guidance.llm = self.llama
         self.vector_store_docs = Chroma(collection_name="docs_collection")
         self.vector_store_convs = Chroma(collection_name="convos_collection")
         tools = {
             "Search Documents": self.search_documents,
             "Search Conversations": self.search_conversations,
         }
         self.andromeda = AndromedaChain(config.andromeda_url)
-        self.document_qa_agent = ChainOfThoughtsAgent(self.andromeda, tools)
+        self.document_qa_agent = ChainOfThoughtsAgent(guidance, tools)
```

Comment:
Rather than overwriting `ChainOfThoughtsAgent`, could we create a new class?

Reply:
Of course! I implemented the class ChainOfThoughtsFlowAgent for the original one